hw/ip/otbn/doc/dv_plan/index.md

```diff
@@ -24,83 +16,124 @@ applicable. Once done, remove this comment before making a PR. -->
-### UVC/agent 2
-[Describe here or add link to its README]
+The top-level testbench is located at `hw/ip/otbn/dv/uvm/tb/tb.sv`.
+This instantiates the OTBN DUT module `hw/ip/otbn/rtl/otbn.sv`.
+
+OTBN has the following interfaces:
+- A [Clock and reset interface]({{< relref "hw/dv/sv/common_ifs#clk_rst_if" >}})
+- A [TileLink interface]({{< relref "hw/dv/sv/tl_agent/README.md" >}}).
+  OTBN is a TL device, which expects to communicate with a TL host.
+  In the OpenTitan SoC, this will be the Ibex core.
+- An idle signal, `idle_o`
+- Two interrupts
+- An [alert interface]({{< relref "hw/dv/sv/alert_esc_agent/README" >}})
+
+The idle and interrupt signals are modelled with the basic
+[`pins_if`]({{< relref "hw/dv/sv/common_ifs#pins_if" >}}) interface.
+
+As well as instantiating OTBN, the testbench also instantiates an `otbn_core_model`.
+This module wraps an ISS (instruction set simulator) subprocess and performs checks to make sure that OTBN behaves the same as the ISS.
+It has a single output signal that goes high when there's a mismatch with the model.
+The testbench wraps this signal with a [`pins_if`]({{< relref "hw/dv/sv/common_ifs#pins_if" >}}) interface.
+
+### otbn_agent
+
+There is a single agent, called `otbn_agent`.
+This is in charge of driving `tl_if`, the TileLink interface to start and stop OTBN.
+It also monitors `alert_if`, `int_if` and `idle_if` for changes in state and `err_if` for any errors reported by the model.
 
 ### Reference models
-[Describe reference models in use if applicable, example: SHA256/HMAC]
+
+The main reference model for OTBN is the instruction set simulator (ISS), which is run as a subprocess by DPI code inside `otbn_core_model`.
+This Python-based simulator can be found at `hw/ip/otbn/dv/otbnsim`.
 
 ### Stimulus strategy
+
+When testing OTBN, we are careful to distinguish between
+
+- behaviour that can be triggered by particular instruction streams
+- behaviour that is triggered by particular external stimuli (register writes; surprise resets etc.)
+
+Testing lots of different instruction streams doesn't really use the UVM machinery, so we have a "pre-DV" phase of testing that generates constrained-random instruction streams (as ELF binaries) and runs a simple block-level simulation on each to check that the RTL matches the model.
+The idea is that this is much quicker for designers to use to sanity-test proposed changes, and can be run with Verilator, so it doesn't require an EDA tool licence.
+This pre-DV phase cannot drive sign-off, but it does use much of the same tooling.
+
+Once we are running full DV tests, we re-use this work, by using the same collection of randomised instruction streams and randomly picking from them for most of the sequences.
+The ELF binaries are all placed in a single directory and named sequentially (`0.elf`, `1.elf`, ..., `1234.elf` or similar).
+This directory and the maximum index are passed to the simulation with plusargs.
+
+<div class="bd-callout bd-callout-warning">
```

Comment:
Are these html tags needed?

Reply:
They do the pretty "pop-out" notes (and I confess that I just copied them from the OTBN ISA doc). I don't *think* you can achieve the same thing with just markdown. If I'm wrong, great! And we should fix the ISA doc to do the same.

Reply:
Maybe @imphil knows?

Reply:
May be worth mentioning here that the design does contain some assertions that aren't included in the Verilator build (currently at least, perhaps SVA support in verilator will reach a point where this is possible). So the UVM DV environment will be checking we're not violating those also on top of the extra external stimuli.

Reply:
No, you can't achieve the same in Markdown and the recommended way by Markdown's original author is to use inline HTML in this case. We could go for a Hugo shortcode if we wanted, but that's the same thing in disguise.

Reply:
@GregAC: Good point about the SV assertions. I'll add a note.

@imphil: I thought that might be the case. Ho hum.